quodlibet/ext/songsmenu/fingerprint/analyze.py

```diff
@@ -172,7 +172,7 @@ def __init__(self, max_workers=None):
     def _get_worker(self):
         """An idle FingerPrintPipeline or None"""
 
-        for worker in self._workers:
+        for worker in self._workers.copy():
```

Comment:
Sets are represented by a hash table and adding or removing items while iterating over it will alter the iteration order. This will cause a RuntimeError.

If you need to modify the set during iteration, it is recommended to iterate over a shallow copy of the set.

For more details you can checkout description of analyzer : https://deepsource.io/gh/ankitdobhal/quodlibet/issue/PTC-W0055/description

Reply:
Yes - but it doesn't look like we're doing any modification here, the `add` is in an `else` block

Reply:
@declension thanks for pointing out , I have fixed it and reverted the change.